crates/service_utils/src/helpers.rs

```diff
@@ -369,3 +373,40 @@ where
 
     response.json::<R>().await
 }
+pub fn generate_snowflake_id(state: &Data<AppState>) -> result::Result<i64> {
+    let mut snowflake_generator = state.snowflake_generator.lock().map_err(|e| {
+        log::error!("snowflake_id generation failed {}", e);
+        result::AppError::UnexpectedError(anyhow!("snowflake_id generation failed {}", e))
+    })?;
+    let id = snowflake_generator.real_time_generate();
+    // explicitly dropping snowflake_generator so that lock is released and it can be acquired in bulk-operations handler
+    drop(snowflake_generator);
+    Ok(id)
+}
+
+pub fn validate_config_tags(
```

Comment:
```suggestion
pub fn parse_config_tags(
```

I think this function is parsing the header value, than just validating

Reply:
No its also validating the regex match. Splitting string and then regex match on each elements